src/main/java/io/netty/incubator/codec/http3/QpackEncoderDynamicTable.java

```diff
@@ -68,9 +67,10 @@ final class QpackEncoderDynamicTable {
 
     /**
      * <a href="https://www.rfc-editor.org/rfc/rfc9204.html#name-maximum-dynamic-table-capac">
-     *     Maximum capacity of the table</a>.
+     *     Maximum capacity of the table</a>. This is set once based on the
+     *     {@link Http3SettingsFrame#HTTP3_SETTINGS_QPACK_MAX_TABLE_CAPACITY} received by the remote peer.
      */
-    private long capacity = -1;
```

Comment:
I believe both capacity variables are required here:
 - Capacity - current capacity of the dynamic table that can be changed multiple times by the peer encoder issuing ["Set Dynamic Table Capacity"](https://www.rfc-editor.org/rfc/rfc9204.html#name-set-dynamic-table-capacity) instruction.  It also needs to be used to check if there is enough space available for entry insertion.
 - Max capacity limits the upper bound of allowed capacity values. The encoder sets its value once based on the peer decoder's settings. It is also used for the `maxEntries` calculation required for encoding of `Required Insert Count`.

Reply:
Hmm... Maybe I am missing something but we never change the dynamic table capacity for the encoder after we set it to the max table capacity received in the SettingsFrame. So I think it is fine ? If we would allow to change the capacity on the fly somehow you would be correct.  I am missing anything ? 

Maybe we could keep both so it would work if we ever allow to change it on the fly. WDYT ?

Reply:
Ah, yes - I've missed the fact that encoder sets the capacity only once (in `QPackEncoder.configureDynamicTable`). What you have in this PR looks about right to me: we can set `capacity` to `maxCapacity` value received from the peer decoder. I also agree that to allow changing encoder dynamic table capacity on the fly both variables would be needed.

For the decoder dynamic table what you have looks right to me:
The `QPackDecoder` tracks both values:
- Max capacity - to precalculate `maxEntries`/`fullRange` variables to decode required insert count from a field line section prefix. And to validate `capacity` value recieved from the peer encoder. 
- Capacity - to check if there is enough space to insert an entry to the dynamic table.

> So I think it is fine ?

Yes, after re-reading the code the change looks fine to me. Thanks.